internal/ethapi/api.go

```diff
@@ -879,9 +900,21 @@ func (s *PublicBlockChainAPI) Call(ctx context.Context, args CallArgs, blockNrOr
 	if err != nil {
 		return nil, err
 	}
-	return result.Return(), nil
+	// If the result contains a revert reason, try to unpack and return it.
+	if len(result.Revert()) > 0 {
+		reason, err := abi.UnpackRevert(result.Revert())
+		if err == nil {
+			return nil, &revertError{
```

Comment:
Now the issue is we have two types `revertError` and `estimateGasError`. Would it be nice to somehow unify the returned error message?

Reply:
I unified them in https://github.com/ethereum/go-ethereum/pull/21083/commits/ce25db796b98452551d8b3598e76292d766db5ba
PTAL